### ISSUE TITLE
Cleanup/deprectate things in `IPython/utils/process.py`

### ISSUE THREAD

**Issue by Carreau** [ipython/ipython#9074]
Lots of things there seem to be used only in tests, and are tools for test. 
If so they should likely be moved to /tests/

Among other `pycmd2argv` seem to be unused/untested. This might need to be deprecated/removed.

**Comment by anantkaushik89** [comment]
Can I take this up if no one is working on it? Can someone please give me any pointers to help me start working on it?

**Comment by willingc** [comment]
@anantkaushik89 Welcome! Thanks for the interest in contributing to IPython. I don't believe that this is being worked on currently.

Let's check with some of the other IPython maintainers to see if this is a good issue to begin with. Currently, the issue seems to need some clarification from the team which tools (if any should be deprecated/removed). Alternatively, there may be another issue better scoped for a new contributor to the project.

@carreau, @takluyver, @minrk, thoughts?

**Comment by Carreau** [comment]
Thanks @willingc for pinging us !

Hi @anantkaushik89 ! As @willingc Welcome !

Of course we can give you pointers ! let us know if it's too detailed, or not enough details, it is often hard to judge if we are explaining right. 

If you look at [this file, and this function](https://github.com/ipython/ipython/blob/69d640651ae812032009e09bfa10f49e6ef076c8/IPython/utils/process.py#L64-L87) it seem to be not used anywhere in our codebase. 

Can you help us check that it is not used ?  It would mean searching through the codebase (with grep for example) and if it's unused we could go 2 ways:

   - Deprecating the function with a deprecation warning. 
   - remove it directly. 

I would go with the second option. 

Once this is done we can go through the other functions in the same file and check whether they are used. 

I'll also try to go through issues and try to find a more fun issue to work on !

**Comment by anantkaushik89** [comment]
Thanks @Carreau ..Yes these pointers really help. I'll go with your suggestion - remove the unused code.

**Comment by anantkaushik89** [comment]
Created PR for the same - https://github.com/ipython/ipython/pull/9864.

**Comment by srinivasreddy** [comment]
@takluyver  Please close this issue

**Comment by takluyver** [comment]
Thanks for the reminder
